src/plugins/mf_priority.cpp

```diff
@@ -0,0 +1,230 @@
+    // treat their first bank as the "default" bank
+    if (users.count (std::atoi (uid)) == 0)
+        users[std::atoi (uid)]["default"] = std::stod (fshare);
+
+    users[std::atoi (uid)][bank] = std::stod (fshare);
+
+    return;
+error:
+    flux_respond_error (h, msg, errno, flux_msg_last_error (msg));
+}
+
+
+/*
+ * Unpack the urgency and userid from a submitted job and call calc_priority (),
+ * which will return a new job priority to be packed.
+ */
+static int priority_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int urgency, userid;
+    char *bank = NULL;
+    int64_t priority;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:i, s{s{s{s?s}}}}",
+                                "urgency", &urgency,
+                                "userid", &userid,
+                                "jobspec", "attributes", "system",
+                                "bank", &bank) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    priority = priority_calculation (p, args, userid, bank, urgency);
+
+    if (flux_plugin_arg_pack (args,
+                              FLUX_PLUGIN_ARG_OUT,
+                              "{s:I}",
+                              "priority",
+                              priority) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_pack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+    return 0;
+}
+
+
+/*
+ * Look up the userid of the submitted job in the multimap; if user is not found
+ * in the map, reject the job saying the user wasn't found in the
+ * flux-accounting database.
+ */
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int userid;
+    char *bank = NULL;
+    std::map<int, std::map<std::string, double>>::iterator it;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s{s{s{s?s}}}}",
+                                "userid", &userid,
+                                "jobspec", "attributes", "system",
+                                "bank", &bank) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    // searching element in std::map by key
+    it = users.find (userid);
+    if (it == users.end ())
```

Comment:
Should `job.validate` also check that the `bank`, if provided, is valid for the given user?

Reply:
Yes, you're right. I just pushed up a `fixup` commit (e92974b) that adds another check in `validate_cb ()` where the passed-in bank is checked under the user, and if it is not found, the job will be rejected with an error message saying that the user does not belong to the passed-in bank. I also added a test (4cc116c) that should trigger this error message.